src/main/java/julia/git/File.java

```diff
@@ -0,0 +1,58 @@
+package julia.git;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+public class File implements Listable {
+    private int size;
+    private final byte[] b;
+
+    File(int size, byte[] b) {
```

Comment:
Now that you're passing the content you don't need to explicitly pass `size`. You can just use `b.length`